src/ansys/aedt/core/modeler/circuits/primitives_twin_builder.py

```diff
@@ -475,3 +475,111 @@ def create_periodic_waveform_source(
+        >>> tb.modeler.schematic.create_component_from_sml(input_file=model, model=model, pins_names=pins_names)
+        >>> tb.release_desktop(False, False)
+        """
+        arg = ["NAME:Options", "Mode:=", 1]
+        arg2 = ["NAME:Models", model + ":=", [True]]
+
+        arg3 = ["NAME:Components", model + ":=", [True, True, model, True, pins_names.lower(), pins_names.lower()]]
+
+        arg.append(arg2)
+        arg.append(arg3)
+        self.o_component_manager.ImportModelsFromFile(input_file, arg)
+        return True
+
+    @pyaedt_function_handler()
+    def update_quantity_value(self, component_name, name, value, netlist_units=""):
+        """Change the property value of a component.
+
+        Parameters
+        ----------
+        component_name : str
+            Component name.
+        name : str
+            Property name.
+        value : str
+            Value of the quantity.
+        netlist_units : str, optional
+            Value of the netlist unit.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        Examples
+        --------
+        >>> from ansys.aedt.core import TwinBuilder
+        >>> tb = TwinBuilder(version="2025.1")
+        >>> G = 0.00254
+        >>> modelpath = "Simplorer Elements\\Basic Elements\\Tools\\Time Functions:DATAPAIRS"
+        >>> source1 = tb.modeler.schematic.create_component("source1", "", modelpath, [20 * G, 29 * G])
+        >>> tb.modeler.schematic.update_quantity_value(source1.composed_name, "PERIO", "0")
+        >>> tb.release_desktop(False, False)
+        """
+        try:
+            self.oeditor.ChangeProperty(
+                [
+                    "NAME:AllTabs",
+                    [
+                        "NAME:Quantities",
+                        ["NAME:PropServers", component_name],
+                        [
+                            "NAME:ChangedProps",
+                            [
+                                "NAME:" + name,
+                                "OverridingDef:=",
+                                True,
+                                "Value:=",
+                                value,
+                                "NetlistUnits:=",
+                                netlist_units,
+                                "ShowPin:=",
+                                False,
+                                "Display:=",
+                                False,
+                                "Sweep:=",
+                                False,
+                                "SDB:=",
+                                False,
+                            ],
+                        ],
+                    ],
+                ]
+            )
+            return True
+        except Exception:  # pragma: no cover
+            self.logger.warning("Property %s has not been edited. Check if readonly.", name)
```

Comment:
```suggestion
            self.logger.warning(f"Property {name} has not been edited. Check if readonly.")
```
We can use f-strings now !!! :D